src/fetch-middleware/with-saleor-domain-present.ts

```diff
@@ -0,0 +1,25 @@
+import { getSaleorHeadersFetchAPI } from "../headers";
+import { createFetchMiddlewareDebug } from "./middleware-debug";
+import { FetchMiddleware } from "./types";
+
+const debug = createFetchMiddlewareDebug("withSaleorDomainPresent");
+
+export const withSaleorDomainPresent: FetchMiddleware = (handler) => async (request) => {
```

Comment:
Should we keep on using it? I believe `domain` is soft-deprecated and we should switch to `saleorApiUrl` instead